### ISSUE TITLE
Unable to show TIF images in Chrome & Firefox

### ISSUE THREAD

**Issue by RSantosh09** [openseadragon/openseadragon#1380]
We have a application where we will show up TIFF images in the IE browser alone till date. 

Now we are checking to support all regular browsers like Chrome & Firefox. But we are unable to show up TIF images in Chrome & Firefox.

Our application is simple and we just need to show up the TIF image with minimal Zooming (IN & OUT) levels.

Now is there any way we can show up TIF images in chrome and Firefox?
We can not afford to change the image format neither we can convert the images at the server side as we have no control over the input images.

**Comment by iangilman** [comment]
This chart confirms that neither Firefox nor Chrome support the TIFF file format:

https://en.wikipedia.org/wiki/Comparison_of_web_browsers#Image_format_support

If you want to use those browsers you will have to convert your images. There are lots of options for doing so. You can write a script to do a one-time conversion on all the files, or you can write a server-side component to convert them when they are ingested.

**Comment by RSantosh09** [comment]
Yes, by converting the image at browser side i can show that image Browser but i need to show that image with OpenSeaDragon so that i can use Some of the OpenSeaDragon features in my application.

So correct me if am wrong, First i need to convert TIF image in server side and then load it in browser using OpenSeaDragon.

Or is there anyway that i can call external script from OpenSeaDragon script and show the converted image using OpenSeaDragon Viewer?

Thanks In Advance!!

**Comment by iangilman** [comment]
Good point... If you have something that can convert a TIFF into a JPEG or PNG on the client side, you certainly can pass that into OSD. Is your image broken into tiles, or is it all in one piece? If it's all one piece, you could make a data URL out of the converted version and pass that to the simple image tile source:

http://openseadragon.github.io/examples/tilesource-image/

**Comment by RSantosh09** [comment]
Yes the process you mentioned above is working for single page image.
I have tried "https://github.com/seikichi/tiff.js" to convert TIFF to PNG and then converted image data URL is binded to OSD viewer. 
Unfortunately for Multi page i couldn't find a work around. so i will work this out in Server side.
Thanks a lot for your support.

**Comment by iangilman** [comment]
Strange... it should work with multi-image as well... What sort of issues are you getting when you try to do so? What does the code look like?

**Comment by RSantosh09** [comment]
Do you mean Multiple Tile Sources to the same OSD viewer (each tile source URL has different image)? If so yes the solution is working fine.

What i meant was, if i have a multi page TIFF image, and am converting the TIFF image at the browser as PNG using external script.
And After conversion am getting single PNG image though i have a multiple page in the actual TIFF image. That converted single PNG image am able to show in the OSD viewer.

Basically am getting a single PNG image after converting the multi page TIFF image at the browser side. So i thought i could handle this Multi page TIFF conversion at server side using my own code or some professional tool.

**Comment by iangilman** [comment]
I see... The conversion tool isn't working with multi-page TIFF. Yes, doing it on the server side might be the right choice then.

**Comment by JerryGit1** [comment]
Hello, I'm sorry to interrupt you. The problem I encountered is very similar to yours. We got a 5G TIF image. The server side is python. Because the image conversion is too big, it is limited by pixels. Do you have any suggestions for converting TIF to jpg?

After conversion, use deep Zoom to cut and then render to the front end

**Comment by iangilman** [comment]
@JerryGit1 Sorry for the late response! I must have missed your question before. 

I recommend using VIPS (you can find the link here: http://openseadragon.github.io/examples/creating-zooming-images/)... It can handle the conversion directly from TIF to DZI without having to do an intermediate step.

**Comment by KempWatson** [comment]
There are a TON of issues using TIFF in browsers. Support is very slim. Where it is supported, it's only a very limited subset of TIFF. BigTIFF is not supported at all. For @JerryGit1 5G image, it CANNOT be TIFF, it must be BigTIFF (and please don't serve a 5GB image to a browser session unless you want the users to commit suicide). Many TIFFs cannot be converted to JPEG, as JPEG is limited to 2 or 4 GB size (TIFF is 4 GB), and JPEG has a limit of 64K pixels/side, TIFF is much higher. Don't use TIFF in browsers unless you want a lot of pain.

**Comment by JerryGit1** [comment]
Thank you very much for your reply. It's settled now. My solution is to use the OpenSeadragon Canvas Overlay HD plug-in. Firstly, the 5G Tiff image is divided into six blocks, and then cut separately. Finally, we use this plug-in to stitch the whole picture. @KempWatson

**Comment by UmangSharma5** [comment]
Can anyone guide me on how to use [GeoTIFFTileSource]  i wanted to render tiff images using openseadragon . Is there a way we can achieve this without converting tiff to png/jpg

**Comment by RhythmAgg** [comment]
@JerryGit1 can you explain how you divided the BigTIFF into blocks as you mentioned before. I have a similar problem and plan on proceeding the same way
